pkg/conf/conf.go

```diff
@@ -256,6 +256,11 @@ func GetPrometheusExporterPorts(pollerName string) (int, error) {
 	if poller == nil {
 		return 0, errors.New(errors.ERR_CONFIG, "Poller does not exist "+pollerName)
 	}
+
+	if Config.Defaults != nil {
```

Comment:
unioning should have already happened

I bet `harvest.go` wasn't converted to use the struct based in this codepath

Reply:
yep that's the case - what if we fix this more globally (so we don't have more examples of this in downstream code) by adding these lines at 173 in conf.go

```
// Load struct based config here so the poller structs are union-ed with defaults
if _, err = GetPollers2(configFp); err != nil {
    return nil, err
}
```

Reply:
I has the same idea first, thanks

Reply:
I tried to use this more globally, changed cmd/harvest/harvest.go to use the struct-based Poller config